src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,232 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+
+import { Alert, AlertVariant } from '@patternfly/react-core/dist/esm/components/Alert';
+import { Button } from '@patternfly/react-core/dist/esm/components/Button';
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Modal, ModalVariant } from '@patternfly/react-core/dist/esm/components/Modal';
+import { TextArea } from '@patternfly/react-core/dist/esm/components/TextArea';
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { Stack } from '@patternfly/react-core/dist/esm/layouts/Stack';
+
+import cockpit from 'cockpit';
+import { FormHelper } from 'cockpit-components-form-helper';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { useInit } from 'hooks';
+import { superuser } from 'superuser';
+
+import "./editor.scss";
+import { useFilesContext } from '../app.tsx';
+import { checkFilename } from '../common.ts';
+import { get_owner_candidates } from '../ownership.tsx';
+
+import { edit_file } from './editor.tsx';
+
+const _ = cockpit.gettext;
+
+async function create_file(filename: string, content?: string, owner?: string) {
+    const file = cockpit.file(filename, { superuser: "try" });
+
+    try {
+        await file.replace("");
+    } catch (err) {
+        console.warn("Cannot create new file", err);
+        throw err;
+    }
+
+    if (owner !== undefined) {
+        try {
+            await cockpit.spawn(["chown", owner, filename], { superuser: "require" });
+        } catch (err) {
+            console.warn("Cannot chown new file", err);
+            throw err;
```

Comment:
In this case it needs to clean up the created file -- it may be dangerous to leave an user-owned file in e.g. a non-writable system directory. The creation already worked, after all.